service/collector/env.go

```diff
@@ -41,19 +57,128 @@ func NewEnv(config EnvConfig) (*Env, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
-	v := &Env{
+	c := &Env{
 		g8sClient: config.G8sClient,
 		logger:    config.Logger,
 	}
 
-	return v, nil
+	return c, nil
 }
 
-func (v *Env) Collect(ch chan<- prometheus.Metric) error {
+func (c *Env) Collect(ch chan<- prometheus.Metric) error {
+	var desiredClusterIDs []string
+	{
+		l, err := c.g8sClient.Core().FlannelConfigs(metav1.NamespaceAll).List(metav1.ListOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, i := range l.Items {
+			desiredClusterIDs = append(desiredClusterIDs, i.Name)
+		}
+	}
+
+	var envClusterIDs []string
+	{
+		files, err := ioutil.ReadDir(envDirectory)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, file := range files {
+			id := clusterIDFromPath(file.Name())
+			if id == "" {
+				return microerror.Maskf(invalidFileError, file.Name())
```

Comment:
IMO this should be `executionError`. It isn't asserted anywhere.

Reply:
Going with `executionFailedError`.